### ISSUE TITLE
SC3028 hostname utility is not in POSIX at all

### ISSUE THREAD

**Issue by polluks** [koalaman/shellcheck#3468]
#### For bugs with existing features

- Rule Id (SC3028):
- My shellcheck version (0.11.0):
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC3028)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
hn=${HOSTNAME}
echo "$hn"
hn=$(HOSTNAME)
echo "$hn"
```

#### Here's what shellcheck currently says:
$(HOSTNAME) is okay

#### Here's what I wanted or expected to see:
use `uname -n`
see https://github.com/gwsw/less/pull/770

**Comment by avih** [comment]
To clarify the issue, shellcheck correctly flags the use of `$HOSTNAME` in a `/bin/sh` script as non-POSIX, and links to https://shellcheck.net/wiki/SC3028 .

However, https://shellcheck.net/wiki/SC3028 recommends to use `$(hostname)` instead.

But the problem is that `hostname` is also not POSIX, which defeats the purpose of fixing a POSIX-compliance issue.

Instead, it could recommend to use `$(uname -n)` - which is POSIX.
